test/testsuites/opsrccreationtests.go

```diff
@@ -48,3 +53,67 @@ func testOperatorSourceGeneratesExpectedObjects(t *testing.T) {
 			helpers.TestOperatorSourceLabelValue,
 			groupGot))
 }
+
+// testRegistryDeploymentRetainsChanges ensures that changes likes annotations mades to the pod spec of the registry
+// deployment associated with an OperatorSource is reatain across updates.
+func testRegistryDeploymentRetainsChanges(t *testing.T) {
+	// Get test namespace
+	namespace, err := test.NewTestCtx(t).GetNamespace()
+	require.NoError(t, err, "Could not get namespace")
+
+	// Check for child resources.
+	err = helpers.CheckChildResourcesCreated(test.Global.Client, helpers.TestOperatorSourceName, namespace, namespace,
+		v1.OperatorSourceKind)
+	require.NoError(t, err)
+
+	client := test.Global.Client
+
+	// Get the registry deployment of the test OperatorSource
+	deployment := getRegistryDeployment(t, helpers.TestOperatorSourceName, namespace)
+	require.NotNil(t, deployment)
+
+	// Add an annotation to the pod template and update the deployment
+	annotationName := "always-here"
+	meta.SetMetaDataAnnotation(&deployment.Spec.Template.ObjectMeta, annotationName, "test")
+	err = client.Update(context.TODO(), deployment)
+	require.NoError(t, err)
+
+	err = helpers.WaitForSuccessfulDeployment(client, *deployment)
+	require.NoError(t, err)
+
+	// Get the test OperatorSource
+	testOpSrc := &v1.OperatorSource{}
+	namespacedName := types.NamespacedName{Name: helpers.TestOperatorSourceName, Namespace: namespace}
+	err = client.Get(context.TODO(), namespacedName, testOpSrc)
+	require.NoError(t, err)
+
+	// Force an update
+	testOpSrc.Status = v1.OperatorSourceStatus{}
+	err = client.Update(context.TODO(), deployment)
```

Comment:
Is this the best way to force an update? Won't this start to fail if we add validation on the status?

Can we just scale the marketplace deployment down to zero and then back up to one instead? Also, that seems like a good utility function to add in general so other tests can reuse it.

Reply:
>Is this the best way to force an update? Won't this start to fail if we add validation on the status?

I agree that it not and we will need to update it with whatever new method we provide to force updates on `OperatorSources`

>Can we just scale the marketplace deployment down to zero and then back up to one instead? Also, that seems like a good utility function to add in general so other tests can reuse it.

I want to target the update to affect just one `OperatorSource` and not restart the operator causing all the objects to be reconciled again.

Reply:
Fair. Let's address it if/when we have that problem. We will probably need to define a method to update just one opsrc at that point anyway.